sopel/modules/url.py

```diff
@@ -36,6 +40,44 @@ class UrlSection(StaticSection):
     exclusion_char = ValidatedAttribute('exclusion_char', default='!')
 
 
+class TitleParser(HTMLParser):
+    def __init__(self):
+        try:
+            HTMLParser.__init__(self, convert_charrefs=False)
+        except TypeError:
+            HTMLParser.__init__(self)
+
+        self.match = False
+        self.in_head = False
+        self.title = ''
+        self.encoding = None
+
+    def handle_starttag(self, tag, attributes):
+        self.match = True if tag == 'title' and self.in_head else False
+        if tag == 'head':
+            self.in_head = True
+
+        # Look for a tag like "<meta http-equiv="Content-Type" content="text/html; charset=utf-8">"
```

Comment:
sure

Reply:
Looks like the updated code never checks at all whether the detected `<meta>` tag is an `http-equiv` or not, let alone whether it's a `Content-Type` tag. There are [several valid (and deprecated) values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta) that aren't charsets.

Probably best to check for the newer `charset` attribute first, then make sure the `<meta>` tag is `http-equiv="content-type"` before parsing its `content`. And once a charset/encoding is found,  break the loop. Ideally, skip all further meta tags once a value is found, since there's no reason to look at any more.